src/main/java/com/example/x3/MovieManagementApp/services/impl/MovieCommentServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package com.example.x3.MovieManagementApp.services.impl;
+
+import com.example.x3.MovieManagementApp.dtos.MovieCommentsDto.MovieCommentAddDto;
+import com.example.x3.MovieManagementApp.dtos.MovieCommentsDto.MovieCommentDto;
+import com.example.x3.MovieManagementApp.entities.MovieComments;
+import com.example.x3.MovieManagementApp.entities.Movies;
+import com.example.x3.MovieManagementApp.repositories.MovieCommentRepository;
+import com.example.x3.MovieManagementApp.repositories.MovieRepository;
+import com.example.x3.MovieManagementApp.services.MovieCommentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class MovieCommentServiceImpl implements MovieCommentService {
+
+    @Autowired
+    MovieCommentRepository movieCommentRepository;
+
+    @Autowired
+    MovieRepository movieRepository;
+
+    @Override
+    public List<MovieComments> findAll() {
+        return movieCommentRepository.findAll();
+    }
+
+    @Override
+    public MovieComments findById(Long id) {
+        Optional<MovieComments> tempMovieComment = movieCommentRepository.findById(id);
+
+        return tempMovieComment.orElse(null);
+    }
+
+    @Override
+    public List<MovieComments> findAllCommentsByMovieId(Long movieId) {
+        Optional<Movies> tempMovie = movieRepository.findById(movieId);
+
+        if (tempMovie.isPresent()) {
+            return movieCommentRepository.findAllByMovieId(movieId);
+        }
+        else {
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public String saveComment(MovieCommentAddDto movieCommentAddDto) {
+        if (movieCommentRepository.findByComment(movieCommentAddDto.getComment()).isPresent()) {
+            return "This comment already exists. Maybe you would like to change it?";
+        }
+
+        MovieComments newMovieComment = new MovieComments();
+        newMovieComment.setComment(movieCommentAddDto.getComment());
+        movieCommentRepository.save(newMovieComment);
+
+        return "Comment is saved";
+    }
```

Comment:
Sorry, I missed this before.  When you go to save the new comment in your MovieCommentServiceImpl, I believe you need to add the rest of the info from your movieCommentAddDto to the newMovieComment object before saving.